src/category_theory/limits/constructions/equalizers.lean

```diff
@@ -76,10 +81,146 @@ end has_equalizers_of_pullbacks_and_binary_products
 open has_equalizers_of_pullbacks_and_binary_products
 /-- Any category with pullbacks and binary products, has equalizers. -/
 -- This is not an instance, as it is not always how one wants to construct equalizers!
-lemma has_equalizers_of_pullbacks_and_binary_products :
+lemma has_equalizers_of_pullbacks_and_binary_products [has_binary_products C] [has_pullbacks C] :
   has_equalizers C :=
 { has_limit := λ F, has_limit.mk
   { cone := equalizer_cone F,
     is_limit := equalizer_cone_is_limit F } }
 
+local attribute[instance] has_pullback_of_preserves_pullback
+
+/-- A functor that preserves pullbacks and binary products also presrves equalizers. -/
+def preserves_equalizers_of_pullbacks_and_binary_products
+    [has_binary_products C] [has_pullbacks C]
+    [preserves_limits_of_shape (discrete.{v} walking_pair) G]
+    [preserves_limits_of_shape walking_cospan.{v} G] :
+  preserves_limits_of_shape walking_parallel_pair G :=
+⟨λ K, preserves_limit_of_preserves_limit_cone (equalizer_cone_is_limit K) $
+{ lift := λ c, begin
+    refine pullback.lift _ _ _ ≫ (@@preserves_pullback.iso _ _ _ _ _ _ _ _).inv,
+    { exact c.π.app walking_parallel_pair.zero },
+    { exact c.π.app walking_parallel_pair.zero },
+    apply (map_is_limit_of_preserves_of_is_limit G _ _ (prod_is_prod _ _)).hom_ext,
+    swap, apply_instance,
+    rintro (_|_),
+    { simp only [binary_fan.mk_π_app_left, category.assoc, ← G.map_comp, prod.lift_fst] },
+    { simp only [binary_fan.mk_π_app_right, category.assoc, ← G.map_comp, prod.lift_snd],
+      exact (c.π.naturality (walking_parallel_pair_hom.left)).symm.trans
+        (c.π.naturality (walking_parallel_pair_hom.right)) },
+  end,
+  fac' := λ c j, begin
+    rcases j with (_|_);
+      simp only [category.comp_id, preserves_pullback.iso_inv_fst, cone.of_fork_π, G.map_comp,
+        preserves_pullback.iso_inv_fst_assoc, functor.map_cone_π_app, eq_to_hom_refl,
+        category.assoc, fork.of_ι_π_app, pullback.lift_fst, pullback.lift_fst_assoc],
+      exact (c.π.naturality (walking_parallel_pair_hom.left)).symm.trans (category.id_comp _)
+  end,
+  uniq' := λ s m h, begin
+    rw iso.eq_comp_inv,
+    have := h walking_parallel_pair.zero,
+    dsimp [equalizer_cone] at this,
+    ext; simpa only [pullback_comparison_comp_fst, pullback_comparison_comp_snd, category.assoc,
+        preserves_pullback.iso_hom, pullback.lift_fst, pullback.lift_snd, category.comp_id,
+        ← pullback_fst_eq_pullback_snd] using this
+  end }⟩
+
+
+-- We hide the "implementation details" inside a namespace
+namespace has_coequalizers_of_pushouts_and_binary_coproducts
```

Comment:
I wonder if you can use `op` to dualize instead of duplicating the code, and whether we can do so in general when the end goal is a Prop so the construction doesn't matter. Are we lacking infrastructure to do so?

Reply:
`is_limit` and `preserves_limits` are not `Prop`s, so it might cause some issues down the road.
But we do have `src/category_theory/limits/opposites` and `src/category_theory/limits/preserves/opposites`, which might be enough should we take the op approach.

Reply:
Thanks for reminding me `is_limit` and `preserves_limits` are not `Prop`s. I see that `has_coequalizers_of_pushouts_and_binary_coproducts` isn't the only end goal here, and you also want the `preserves_` statement. I suspect since the `has_` hypotheses are used to construct the `preserves_`, maybe the explicit construction doesn't matter after all, but maybe I'm wrong.

Reply:
Is either of you considering the refactor to make `preserves_limits` and friends `Prop`s?

Reply:
I think `preserves_limits` carrying data has never caused me any problems. So I probably won't attempt the refactor in the near future.